recipes/CommonVoice/ASR/transformer/hparams/train_ar_hf_whisper.yaml

```diff
@@ -0,0 +1,153 @@
+train_log: !ref <output_folder>/train_log.txt
+
+# URL for the biggest Fairseq english whisper model.
+whisper_hub: openai/whisper-tiny
+language: arabic #Language for whsiper tokenizer. It should be compatible with <locale> 
+test_only: False # Set it to True if you only want to  do the evaluation
+
+# Normalize the english inputs with
+# the same normalization done in the paper
+normalized_transcripts: True
+
+# Data files
+locale: ar # use 'it' for italian, 'fr' for french, 'en' for english , It is alnaguge for common-voice data. It should be compatible with <language> 
+data_folder: !ref 'data/common_voice_10_0/<locale>' 
+train_tsv_file: !ref <data_folder>/train.tsv  # Standard CommonVoice .tsv files
+dev_tsv_file: !ref <data_folder>/dev.tsv  # Standard CommonVoice .tsv files
+test_tsv_file: !ref <data_folder>/test.tsv  # Standard CommonVoice .tsv files
+accented_letters: True
+train_csv: !ref <save_folder>/train.csv
+valid_csv: !ref <save_folder>/dev.csv
+test_csv: !ref <save_folder>/test.csv
+skip_prep: False # Skip data preparation
+
+# We remove utterance slonger than 10s in the train/dev/test sets as
+# longer sentences certainly correspond to "open microphones".
+avoid_if_longer_than: 10.0
+
+ckpt_interval_minutes: 30 # save checkpoint every N min
+
+# Training parameters
+number_of_epochs: 1
+lr_whisper: 0.00003
+sorting: ascending
+auto_mix_prec: False
+sample_rate: 16000
+
+ignore_index: -100  # For cross-entropy loss
+label_smoothing: 0
+
+
+# With data_parallel batch_size is split into N jobs
+# With DDP batch_size is multiplied by N jobs
+batch_size: 12
+test_batch_size: 8
+
+# These values are only used for the searchers.
+# They needs to be hardcoded and should not be changed with Whisper.
+# They are used as part of the searching process.
+# The bos token of the searcher will be timestamp_index
+# and will be concatenated with the bos, language and task tokens.
+timestamp_index: 50363
+eos_index: 50257
+bos_index: 50258
+
+# Decoding parameters
+min_decode_ratio: 0.0
+max_decode_ratio: 0.1
+test_beam_size: 8
+
+# Model parameters
+freeze_whisper: False
+freeze_encoder: True
+
+
+
+train_loader_kwargs:
+    batch_size: !ref <batch_size>
+
+valid_loader_kwargs:
+    batch_size: !ref <batch_size>
+
+test_loader_kwargs:
+    batch_size: !ref <test_batch_size>
+
+
+#
+# Functions and classes
+#
+epoch_counter: !new:speechbrain.utils.epoch_loop.EpochCounter
+    limit: !ref <number_of_epochs>
+
+augmentation: !new:speechbrain.lobes.augment.TimeDomainSpecAugment
+    sample_rate: !ref <sample_rate>
+    speeds: [95, 100, 105]
+
+whisper: !new:speechbrain.lobes.models.huggingface_whisper.HuggingFaceWhisper
+    source: !ref <whisper_hub>
+    freeze: !ref <freeze_whisper>
+    freeze_encoder: !ref <freeze_encoder>
+    save_path: !ref <save_folder>/whisper_checkpoint
+    encoder_only: False
+
+log_softmax: !new:speechbrain.nnet.activations.Softmax
+    apply_log: True
+
+ce_loss: !new:torch.nn.CrossEntropyLoss
```

Comment:
Why using cross entropy loss directly ? Why not using the NLL from SpeechBrain ?

Reply:
I need to add ignore index  for ignoring padded tokens while calculating loss and speechbrain NLL doesn't have that. Instead of changing the function , I use the CrossEntropyLoss which do the same thing and neater.

Reply:
Oh, You are right, I forgot about it.  will re-run one of my experiment with the correct version to see how it affect the result.

Reply:
I have re-run the experiment with correct loss and with large-v2 model instead of large. I am uploading them on Huggingface now. Once done, I will update the readmefine with the HF link(probably by tomorrow)

Reply:
I have uploaded all models on HF and committed the final changes. I am still uploading the models on Gdrive but it might take longer but I put the correct link on both HF and readme file.